### PR TITLE
Add fails-to-deliver data

### PR DESCRIPTION
**Pull request by hinxx**
First attempt at providing fails-to-deliver data from SEC as proposed in #362.
Load ticker, go into `dd` menu, use `ftd` command.
```
(✨) (dd)> ftd -h
usage: ftd [-n N_NUM] [-h]

Prints latest fails-to-deliver data. [Source: SEC]

optional arguments:
  -n N_NUM, --num N_NUM
                        number of latest fails-to-deliver being printed
  -h, --help            show this help message
```

By default return 10 latest dates with FTD data.
```
(✨) (dd)> ftd
       Date  Quantity
 2021-03-31     14031
 2021-03-30     17841
 2021-03-29     86859
 2021-03-26     62109
 2021-03-25     20295
 2021-03-24     38387
 2021-03-23     83058
 2021-03-22     17163
 2021-03-19       637
 2021-03-18     32220

```

No support for custom start date as it complicates things for lazy me today.

### CODE REVIEW

File: gamestonk_terminal/due_diligence/sec_view.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            0 : ns_parser.n_num
+        ]
+        ftds_data["SETTLEMENT DATE"] = ftds_data["SETTLEMENT DATE"].apply(
+            lambda x: datetime.strptime(str(x), "%Y%m%d").strftime("%Y-%m-%d")
+        )
+        print(
+            ftds_data.rename(
+                columns={"SETTLEMENT DATE": "Date", "QUANTITY (FAILS)": "Quantity"}
+            ).to_string(index=False)
+        )
```
Comment (DidierRLopes) [inline gamestonk_terminal/due_diligence/sec_view.py:90]: Missing a print("")

**Review by DidierRLopes — APPROVED** [review]
Amazing! The only thing I'd say is to by default do a plot bar, as it is easier to visualise than by looking at the numbers!

You can copy the plt.bar code from this PR:
https://github.com/DidierRLopes/GamestonkTerminal/pull/363

I can add it myself if you prefer, let me know!

Also, you can add a line of code on the "ROADMAP" with: 
```
* [x] Add failure to deliver command (@hinxx) - [PR #366](#https://github.com/DidierRLopes/GamestonkTerminal/pull/366)
```

Amazing work!

File: gamestonk_terminal/due_diligence/sec_view.py (hunk 1 above)
Comment (hinxx) [reply gamestonk_terminal/due_diligence/sec_view.py:90]: should be on line 86 :)

**Comment by hinxx** [comment]
> Amazing! The only thing I'd say is to by default do a plot bar, as it is easier to visualise than by looking at the numbers!

Let me try to add this from the PR you mentioned!

File: gamestonk_terminal/due_diligence/sec_view.py (hunk 1 above)
Comment (DidierRLopes) [reply gamestonk_terminal/due_diligence/sec_view.py:90]: I meant an additional one, to give a new line between last print and new command.

Captura de ecrã 2021-04-18, às 23 33 24

**Comment by hinxx** [comment]
I noticed that using a split view/model would be nice to have here, too. Let me do that as well before you merge this.

Screenshot_2021-04-19_01-02-41

File: gamestonk_terminal/due_diligence/sec_view.py (hunk 1 above)
Comment (hinxx) [reply gamestonk_terminal/due_diligence/sec_view.py:91]: :facepalm:, gotcha!
Mmm, do you want to have console output and the bar plot?

**Comment by DidierRLopes** [comment]
> I noticed that using a split view/model would be nice to have here, too. Let me do that as well before you merge this.
> 
> Screenshot_2021-04-19_01-02-41

Really nice!! I thought about that straight after I reviewed it.

To improve the plot, I would do 2 things:
* Change ylabel to Shares [K], and when doing plt.bar I would divide it by 1_000. It seems a better unit for this purpose.
* Add spacing on the xticks, you can do this:
```
import matplotlib.dates as mdates

plt.gca().xaxis.set_major_formatter(mdates.DateFormatter("%Y/%m/%d"))
plt.gca().xaxis.set_major_locator(mdates.DayLocator(interval=7))
```

File: gamestonk_terminal/due_diligence/sec_view.py (hunk 1 above)
Comment (DidierRLopes) [reply gamestonk_terminal/due_diligence/sec_view.py:91]: Nah, I don't think is needed. That dataframe is the one that your model should output, and the view digest :)

**Review by aia — APPROVED** [review]

**Pull request merged**